ci/fetch-artifacts.sh

```diff
@@ -2,48 +2,92 @@
 
 # script to fetch artifacts from Win,OSX CI to Travis,
 # where all 3 files are deployed to GitHub for release. 
+# The curent pwd should be build/scripts when running this script.
 
-WINDOWS_ARTIFACT_URL="https://ci.appveyor.com/api/buildjobs/by4cpqmul79tp9ag/artifacts/build/scripts/htm_core-6866bd13ece35608313709e82c29b2260ee7be73-windows64.zip" #FIXME make the URLs automated, needs API?
-OSX_ARTIFACT_URL="https://3238-118031881-gh.circle-artifacts.com/0/tmp/uploads/htm_core-7b7b53c798b3795088ce772e96c6994068fc1302-darwin64.tar.gz"
+
+export VERSION=`cat ../../VERSION`
+echo "Released version: $VERSION"
+echo
+
+export TRAVIS_FILE="htm_core-$VERSION-linux64.tar.gz"
+
+#WINDOWS_ARTIFACT_URL="https://ci.appveyor.com/api/buildjobs/by4cpqmul79tp9ag/artifacts/build/scripts/htm_core-6866bd13ece35608313709e82c29b2260ee7be73-windows64.zip?branch=master" #FIXME make the URLs automated, needs API?
+#OSX_ARTIFACT_URL="https://3238-118031881-gh.circle-artifacts.com/0/tmp/uploads/$CIRCLE_FILE"
+
+
+echo
+echo "Fetching Windows artifacts:"
+export APPVEYOR_TOKEN="v2.5tido3qf80n33le5bi3e"  # token by David Keeney.  TODO: replace with project token.  Needs Admin permissions.
+export APPVEYOR_FILE="htm_core-$VERSION-windows64.zip"
+export APPVEYOR_ARTIFACT_URL="https://ci.appveyor.com/api/projects/htm-community/htm.core/artifacts/build/scripts/$APPVEYOR_FILE?branch=master" 
 
 echo "Fetching Windows artifacts:" 
-retry=false
+# Perform authentication
+
 for i in `seq 10`; do
-  echo "Downloading artifacts. Try $i"
-  wget "$WINDOWS_ARTIFACT_URL" || retry=true
-  if [[ x"$retry" == "xtrue" ]]; then
-    echo "sleep for 5min is good"
-    sleep 300 # wait 5min to retry. Other CI needs to finish its build
-    retry=false
+  echo "  Downloading artifacts. Try number $i"
+  # get the headers from the query for the latest master build.  
+  # This is most likely a redirect to the real URL of the artifact.
+  # So, Locate the line for "location:" and use the word in quotes as the real URL.
+  # Then we can use the -O option for the filename of the file it found.
+  page=`curl -v -X GET -I -H "Authorization: Bearer $APPVEYOR_TOKEN" $APPVEYOR_ARTIFACT_URL 2>/dev/null`
```

Comment:
this is smart! But isn't there an API way to simply download the artifacts while providing the token etc? (And if you find one, please add a web-page with the source to the doc here)